packages/react-core/src/components/TreeView/TreeView.tsx

```diff
@@ -87,7 +87,7 @@ export const TreeView: React.FunctionComponent<TreeViewProps> = ({
     <TreeViewList isNested={isNested} onSearch={onSearch} searchProps={searchProps}>
       {data.map(item => (
         <TreeViewListItem
-          key={item.name.toString()}
+          key={item.id.toString()}
```

Comment:
The only issue I see here, is that the id is not a required field, but the name is a required field on the TreeViewDataItem. So maybe it can use the id unless it's undefined? And if so, then use the name?